packages/portfolio-contract/resolver-service.md

```diff
@@ -0,0 +1,140 @@
+# Resolver Service
+
+## Overview
+
+The resolver service monitors and resolves transactions between Agoric and remote EVM chains."Resolving" implies determining and reporting the transaction status (SUCCESS or FAILED) back to the ymax contract. It uses WebSocket connections to listen for specific events on EVM chains and automatically marks transactions as resolved when the expected events are detected.
+
+## How It Works
+
+The resolver:
+- Maintains active WebSocket connections to EVM chains
+- Listens for specific contract events based on transaction type
+- Automatically resolves transactions when expected events are detected
+- Sends alerts when transactions cannot be resolved automatically
+- Never gives up on a transaction unless manually resolved
+
+## Supported Transaction Types
+
+### 1. MAKE_ACCOUNT
+
+**Purpose:** Creates a remote EVM wallet for a user.
+
+**How it resolves:**
+- Listens for `SmartWalletCreated` events from the [Factory contract](https://github.com/agoric-labs/agoric-to-axelar-local/blob/c5b5b2892fe4fe3f822ba460dc9b35239a3fdc2e/packages/axelar-local-dev-cosmos/src/__tests__/contracts/Factory.sol#L155)
+- Automatically resolves when event matches:
+  - **Wallet address:** The created wallet address matches the expected address
+
+**Limitations:**
+- Cannot detect failures automatically
+- Failures require manual resolution
```

Comment:
Something about "if  X hasn't happened in over Y minutes, something is probably wrong" would be nice. Not explicitly about alerts, but suggestive of how we set up our alerts.